### ISSUE TITLE
1.0 Release

### ISSUE THREAD

**Issue by sagebind** [sagebind/isahc#25]
This is a master issue for tracking the upcoming 1.0 release of Isahc.

Isahc's API has pretty much settled into a comfortable position at this point and has no outstanding obvious API deficiencies any more, so now is a great time to make the few pending breaking changes we want to make and release the next version as 1.0.

I've decided that we will _not_ wait for our public dependencies to reach 1.0 first even though it is recommended, because we have very few such dependencies and they move slowly anyway.

### Timeline

1.0 is scheduled to be released before the end of December 2020.

### API

- [x] Make changes to error types: #182
- [x] Split sync and async responses to avoid existing confusion: #202
- [x] Determine whether the body type ought to change further
- [x] Review serialization/deserialization ergonomics. We don't need to implement #211 as the current methods can be changed to convenience methods around a more powerful API later, but we should make sure that the existing methods make sense with that in mind.
- [x] Fix our cookie API situation: #215
- [x] Pin minimum supported Rust version (MSRV) to a specific version. Figure out which version that can be. #219

### Release tasks

- [x] Write blog post announcement
- [x] Publish to Crates.io

---

<details>
<summary>Original description</summary>

Isahc 1.0 will encompass a stabilization of the API, and will include full `Future` and async/await support. The plan is to release a new minor version with the updated API after futures are stabilized in the standard library. After that, let the API soak for a month leaving room for any fixes that might be needed or changes that need to be made to the design. After that we'll release version 1.0.

Most of the open feature requests will be deferred to post-1.0 (though the API needs to keep these requests in mind). The things that must be done before 1.0 are tracked in this milestone: https://github.com/sagebind/isahc/milestone/3

Preparation for 1.0 will be done on a separate branch from `master`, since Isahc is already a "stable" library and requiring nightly is not on the table.
</details>

**Comment by sagebind** [comment]
It would be preferable to wait for the `http` crate to stabilize first, since it is part of the API, according to the API guidelines: https://rust-lang-nursery.github.io/api-guidelines/print.html#public-dependencies-of-a-stable-crate-are-stable-c-stable

**Comment by sagebind** [comment]
Note to self: 1.0-beta.1 will be scheduled for December 12, if everything goes as planned. We will release at least 1 beta to be used early to ensure everything is ready and the new API designs are working as intended.

**Comment by sagebind** [comment]
This is now complete! Check out the [official announcement post](https://github.com/sagebind/isahc/discussions/282).
